### ISSUE TITLE
Consolidate SPI usage 

### ISSUE THREAD

**Issue by ldelossa** [cilium/cilium#34790]
Currently, the SPI value associated with an IPsec SA is populated across 3 maps. 

1. The tunnel map
2. The node map
3. The IPcache 

Ideally, we collapse the SPI value to only be populated in the node map as its property best associated with a specific Node (in our IPsec model). 

This should be done as the starting point for https://github.com/cilium/cilium/issues/20170

**Comment by jschwinger233** [comment]
Also in encrypt_map used in get_min_encrypt_key(): https://github.com/cilium/cilium/blob/9c5c71a34eb09f15a77fc4dc3f814f7847e2af17/bpf/lib/encrypt.h#L36

**Comment by pchaigno** [comment]
Unless we add information for the local node to the node map, we'd have to keep the encrypt map. The maps cited by Louis contain SPIs for remote nodes, whereas the encrypt map contains the SPI for the local node.

**Comment by julianwiedmann** [comment]
Would the first step be to set `use_spi_from_map = true` for more / all callers of `set_ipsec_encrypt()` ?

**Comment by ldelossa** [comment]
@julianwiedmann yes, I don't immediately see why we can't grab the SPI and NodeID at once moving forward.

**Comment by pchaigno** [comment]
We can. We're not doing it today for historic reasons. Cilium IPsec was initially built to allow per-pod encryption enablement, but that was never actually used/exposed to users. So having per-node SPI, as would be the case with the node ID map, should be okay.

**Comment by julianwiedmann** [comment]
> Currently, the SPI value associated with an IPsec SA is populated across 3 maps.
> 
>     1. The tunnel map
> 
>     2. The node map
> 
>     3. The IPcache
> 
> 
> Ideally, we collapse the SPI value to only be populated in the node map as its property best associated with a specific Node (in our IPsec model).
> 
> This should be done as the starting point for [#20170](https://github.com/cilium/cilium/issues/20170)

As an update to ☝️:
1. the tunnel map is gone: https://github.com/cilium/cilium/pull/38839
2. from a datapath PoV, the node map is now the source of truth for a remote endpoint's SPI: https://github.com/cilium/cilium/pull/41525

So as discussed in https://github.com/cilium/cilium/pull/41525, we are in a position where the ipcache entries no longer need to know about the detailed SPI values. (@smagnani96).
